src/ert/storage/migration/to9.py

```diff
@@ -49,22 +66,25 @@ def _migrate_response_configs_wrt_finalized_keys(path: Path) -> None:
                         if config["has_finalized_keys"]:
```

Comment:
This would fail with `KeyError` should it not be the case

Reply:
Fixed the logic and added so this case is covered